src/site/content/en/blog/nfc/index.md

```diff
@@ -0,0 +1,911 @@
+
+```js
+function readTextRecord(record) {
+  console.assert(record.recordType === "text");
+  const textDecoder = new TextDecoder(record.encoding);
+  console.log(`Text: ${textDecoder.decode(record.data)} (${record.lang})`);
+}
+```
+
+Writing a simple text record can be done by passing a string to the NDEFWriter
+`write()` method.
+
+```js
+const writer = new NDEFWriter();
+await writer.write("Hello World");
+```
+
+Text records are UTF-8 by default and assume the current document's language but
+both properties (`encoding` and `lang`) can be specified using the full syntax
+for creating a custom NDEF record.
+
+```js
+function a2utf16(string) {
+  let result = new Uint16Array(string.length);
+  for (let i = 0; i < string.length; i++) {
+    result[i] = string.codePointAt(i);
+  }
+  return result;
+}
+
+const textRecord = {
+  recordType: "text",
+  lang: "fr",
+  encoding: "utf-16",
+  data: a2utf16("Bonjour, François !")
+};
+
+const writer = new NDEFWriter();
+await writer.write({ records: [textRecord] });
+```
+
+### Read and write a URL record
+
+The URL record's `data` can be decoded with a `TextDecoder`.
+
+```js
+function readUrlRecord(record) {
+  console.assert(record.recordType === "url");
+  const textDecoder = new TextDecoder();
+  console.log(`URL: ${textDecoder.decode(record.data)}`);
+}
+```
+
+To write a URL record, pass an NDEF message dictionary to the NDEFWriter
+`write()` method. The URL record contained in the NDEF message is defined as an
+object with a `recordType` key set to `"url"` and a `data` key set to the URL
+string.
+
+```js
+const urlRecord = {
+  recordType: "url",
+  data:"https://w3c.github.io/web-nfc/"
+};
+
+const writer = new NDEFWriter();
+await writer.write({ records: [urlRecord] });
+```
+
+### Read and write a MIME type record
+
+The `mediaType` property of a MIME type record represents the MIME type of the
+NDEF record payload so that `data` can be properly decoded. For instance, use
+JSON.parse to decode JSON text and an Image element to decode image data.
+
+```js
+function readMimeRecord(record) {
+  console.assert(record.recordType === "mime");
+  if (record.mediaType === "application/json") {
+    const textDecoder = new TextDecoder();
+    console.log(`JSON: ${JSON.parse(decoder.decode(record.data))}`);
+  }
+  else if (record.mediaType.startsWith('image/')) {
+    const blob = new Blob([record.data], { type: record.mediaType });
+    const img = new Image();
+    img.src = URL.createObjectURL(blob);
+    document.body.appendChild(img);
+  }
+  else {
+    // TODO: Handle other MIME types.
+  }
+}
+```
+
+To write a MIME type record, pass an NDEF message dictionary to the NDEFWriter
+`write()` method. The MIME type record contained in the NDEF message is defined
+as an object with a `recordType` key set to `"mime"`, a `mediaType` key set to
+the actual MIME type of the content, and a `data` key set to an object that can
+be either an ArrayBuffer or provides a view on to an ArrayBuffer (e.g.
```

Comment:
ArrayBuffer should be formatted so: `ArrayBuffer`.

Reply:
Done